### DISCUSSION TITLE
New library

### DISCUSSION THREAD

**Discussion by BunchOfBits** [logisim-evolution/logisim-evolution#1654] [General]
I intend to add a new library to LE, containing relevant Am29xx chips (Am2901, 02, 04, 09, 10 etc). Although no longer commercially available for a very long time, I think they still present interesting building blocks for basic CPU design. Before I spend the time to develop these models I would like to know if there would be any objections to adding them to LE.

**Comment by R3dst0ne** [comment]
If you think they are useful, feel free to create them.
If they get merged into LSe is another question. IMO, only components that are useful for a majority of the user base should be merged.
The good thing about LSe is that you can use external jar libraries, so even if it ends up not being merged, you can still release and use it as an external jar library.

**Reply by BunchOfBits** [reply]
I don't intend to create, publish and maintain separate libraries outside of the LE project. Libraries like that likely end up not being used at all, simply because nobody knows about their existence: "onbekend maakt onbemind" is a Dutch saying which comes to mind ;-) (I really don't know how to properly translate that into English)
I'm simply testing the water here: would there be an interest in such a library in LE or not. I'm perfectly happy with either yes or no, I really don't mind. I just think that these chips, in spite of their venerable age, can still aid in teaching the basics of CPU design (have a look at the Wikipedia page about them and you will find many famous brands, Digital, Apollo, etc which used these chips). If there is simply no interest in such a library then I will try to contribute to LE on another topic, I just want to spend my time wisely.

**Reply by BFH-ktt1** [reply]
For my part, I do not have a problem that we add a new library to the set. We can have it as "hidden" library, if the list of libraries become too long, or it is not used that much. As long as there is a use case for them I'm very happy with new libraries. The basic template can "render them invisible", and a user template can make them visible. I agree with "onbekend maakt onbemind" and want to add "Bruikbaar en zelden ingezet is beter dan het weg laten en potentiele gebruikers verliezen", use deepl to get the meaning :-) There is just one point which is important, that is the maintenance of the library, I have seen that over the last years, improvements in LE broke components; adding more make the chance that things get broken in the future bigger. We have to find a solution to this problem, but this has only sideways to do with your request. 
my 5c's and I'm looking forward to PR's!

**Reply by maehne** [reply]
I agree completely with @BFH-ktt1 on this and look also forward to your next PRs! We are in general very open to any contributions, which can benefit the Logisim-evolution users! They shall have the opportunity to shape the software according to their needs. We just want to ensure that the overall functionality and code quality does not deteriorate. 

We would also like to see contributions to the documentation and help system. Personally, I feel that is an area, where we have been falling behind, as newer features haven't been documented as thoroughly or at all, as those which were already present in the classic Logisim.

**Reply by BunchOfBits** [reply]
Ok, thanks for the feedback. I will start working on it and look forward to your PR feedback
